src/Durable/DurableController.cs

```diff
@@ -0,0 +1,143 @@
+                    inputData.First(item => item.Name == _durableFunctionInfo.OrchestrationClientBindingName)
+                        .Data.ToObject();
+
+                _powerShellServices.SetOrchestrationClient(orchestrationClient);
+            }
+            else if (_durableFunctionInfo.IsOrchestrationFunction)
+            {
+                ThrowIfDurableNotEnabled();
+
+                _orchestrationBindingInfo = CreateOrchestrationBindingInfo(inputData);
+                _powerShellServices.SetOrchestrationContext(_orchestrationBindingInfo.Context);
+            }
+        }
+
+        public void AfterFunctionInvocation()
+        {
+            _powerShellServices.ClearOrchestrationContext();
+        }
+
+        public bool TryGetInputBindingParameterValue(string bindingName, out object value)
+        {
+            if (_orchestrationInvoker != null
+                && _orchestrationBindingInfo != null
+                && string.CompareOrdinal(bindingName, _orchestrationBindingInfo.ParameterName) == 0)
+            {
+                value = _orchestrationBindingInfo.Context;
+                return true;
+            }
+
+            value = null;
+            return false;
+        }
+
+        public void AddPipelineOutputIfNecessary(Collection<object> pipelineItems, Hashtable result)
+        {
+            var shouldAddPipelineOutput =
+                _durableEnabled && _durableFunctionInfo.Type == DurableFunctionType.ActivityFunction;
+
+            if (shouldAddPipelineOutput)
+            {
+                var returnValue = FunctionReturnValueBuilder.CreateReturnValueFromFunctionOutput(pipelineItems);
+                result.Add(AzFunctionInfo.DollarReturn, returnValue);
+            }
+        }
+
+        public bool TryInvokeOrchestrationFunction(out Hashtable result)
+        {
+            if (!_durableFunctionInfo.IsOrchestrationFunction)
+            {
+                result = null;
+                return false;
+            }
+
+            result = _orchestrationInvoker.Invoke(_orchestrationBindingInfo, _powerShellServices);
+            return true;
+        }
+
+        private static OrchestrationBindingInfo CreateOrchestrationBindingInfo(IList<ParameterBinding> inputData)
+        {
+            // Quote from https://docs.microsoft.com/en-us/azure/azure-functions/durable-functions-bindings:
+            //
+            // "Orchestrator functions should never use any input or output bindings other than the orchestration trigger binding.
+            //  Doing so has the potential to cause problems with the Durable Task extension because those bindings may not obey the single-threading and I/O rules."
+            //
+            // Therefore, it's by design that input data contains only one item, which is the metadata of the orchestration context.
+            var context = inputData[0];
+
+            return new OrchestrationBindingInfo(
+                context.Name,
```

Comment:
The only way for this to happen is if the Durable extension is broken so that it does not send the orchestration context data to the orchestrator function. In this case, many things will break, so this will be noticed before this line of code is executed. This cannot be caused by a user error.